test/IntegrationTests/ModuleTests.cs

```diff
@@ -0,0 +1,58 @@
+// <copyright file="ModuleTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.IO;
+using System.Threading.Tasks;
+using IntegrationTests.Helpers;
+using Newtonsoft.Json;
+using VerifyXunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace IntegrationTests;
+
+[UsesVerify]
+public class ModuleTests : TestHelper
+{
+    public ModuleTests(ITestOutputHelper output)
+        : base("Modules", output)
+    {
+    }
+
+    [Fact]
+    public async Task RunApplication()
+    {
+        var tempPath = Path.GetTempFileName();
+
+        RunTestApplication(arguments: $"--temp-path {tempPath}");
+
+        if (!File.Exists(tempPath))
+        {
+            Assert.Fail("Could not find modules report file.");
+        }
+
+        var json = File.ReadAllText(tempPath);
+        var modules = JsonConvert.DeserializeObject<string[]>(json);
+
+        // Cleanup
+        File.Delete(tempPath);
+
+        await Verifier.Verify(modules)
+           .UniqueForOSPlatform()
+           .UniqueForRuntime()
```

Comment:
We miss the distinction between `netcoreapp3.1` and `net6.0`. Right now we have the same output. Is it OK? Right now we have the exact same output for all configuration so in theory we could even have on `verified.txt` file. When I look at the test I think having a single file would be good enough...

Reply:
net3.1 and net6.0 are same, NET Fx is different.

Reply:
```csharp
            await Verifier.Verify(modules)
#if NETFRAMEWORK
                 .UniqueForRuntime()
#endif
                 .DisableDiff();
```

I would have preferred this, but .NET Core somehow gets into conflict with existing .NET Fx verify file 🙄

Reply:
I think we can simply have ` await Verifier.Verify(modules).DisableDiff();`.

Reply:
Not really, then .NET Fx will pick up .NET Core verify files.

Reply:
I have not noticed the difference 😂 how about 

```csharp
            await Verifier.Verify(modules)
                 .UniqueForRuntime()
                 .DisableDiff();
```

?

Reply:
> net3.1 and net6.0 are same, NET Fx is different.

I have missed this comment...

Reply:
Testing another option that reduces verify files to only 2.

```csharp
            await Verifier.Verify(modules)
#if NETFRAMEWORK
                .UseFileName("ModuleTests.RunApplication.NetFx")
#else
                .UseFileName("ModuleTests.RunApplication.NetCore")
#endif
                .DisableDiff();
```